test/integration/protocol-profile-behavior/sanity.test.js

```diff
@@ -368,6 +368,281 @@ describe('protocolProfileBehavior', function () {
         });
     });
 
+    describe('with noCacheHeader: true', function () {
```

Comment:
these tests are added in the wrong test suite.
Also, why are these tested with redirects?